src/webhooks/index.ts

```diff
@@ -66,6 +68,65 @@ export default function (
             (event.data.object as Stripe.PaymentMethod).id,
           );
           break;
+
+        case 'setup_intent.succeeded': {
+          let customerId: string;
+          const coupon = (event.data.object as Stripe.SetupIntent).metadata?.coupon
+            ? {
+                coupon: (event.data.object as Stripe.SetupIntent).metadata?.coupon as string,
+              }
+            : undefined;
+
+          try {
+            const getUser: User = await usersService.findUserByUuid(
+              (event.data.object as Stripe.SetupIntent).metadata?.uuid as string,
+            );
+
+            const updateCustomer: Stripe.Response<Stripe.PaymentMethod> = await stripe.paymentMethods.attach(
+              (event.data.object as Stripe.SetupIntent).payment_method as string,
+              {
+                customer: getUser.customerId,
+              },
+            );
+
+            customerId = updateCustomer.customer as string;
+          } catch (err) {
+            const customer: Stripe.Customer = await stripe.customers.create({
+              name: (event.data.object as Stripe.SetupIntent).metadata?.name,
+              email: (event.data.object as Stripe.SetupIntent).metadata?.email,
+              payment_method: (event.data.object as Stripe.SetupIntent).payment_method as string,
+            });
+
+            customerId = customer.id;
+          }
+
+          await stripe.subscriptions.create({
+            customer: customerId,
+            default_payment_method: (event.data.object as Stripe.SetupIntent).payment_method as string,
+            items: [
+              {
+                price: (event.data.object as Stripe.SetupIntent).metadata?.priceId as string,
+                metadata: {
+                  is_teams: 0,
+                },
+              },
+            ],
+            expand: ['latest_invoice.payment_intent'],
+            ...coupon,
+          });
+
+          await handleSetupIntentCompleted(
+            event.data.object as Stripe.SetupIntent,
+            usersService,
+            paymentService,
+            fastify.log,
+            cacheService,
+            config,
+            customerId,
```

Comment:
A better idea could be to pass the `customer` directly

Reply:
We can't pass the object directly. For example, we can't use the deleted param when the user exists.